src/common/framework/test_group.ts

```diff
@@ -120,10 +123,27 @@ class TestBuilder<F extends Fixture, P extends {}> {
   }
 
   fn(fn: TestFn<F, P>): void {
+    assert(this.testFn === undefined);
     this.testFn = fn;
   }
 
-  checkCaseNamesAndDuplicates(): void {
+  unimplemented(): void {
+    assert(this.testFn === undefined);
+    this.testFn = () => {
+      throw new SkipTestCase('test unimplemented');
+    };
+  }
+
+  validate(): void {
+    const testPathString = this.testPath.join(kPathSeparator);
+    assert(this.testFn !== undefined, () => {
+      let s = `Test is missing .fn(): ${testPathString}`;
+      if (this.testConstructorStack.stack) {
+        s += `\n-> test created at:\n${this.testConstructorStack.stack}`;
```

Comment:
at least in Chrome, i think this will say: 
> -> test created at:
> Error
> ___ at ...
> ___ at ...

I remember you may have some utilities elsewhere that separate the "Error" message and the stack ?

Reply:
yes, in chrome it will. I think I used to have utilities to do that but they just stripped the first line from the message, and that doesn't actually work (message could have a newline in it, some engines don't put a line at the top of .stack, etc.)